packages/experimental/src/smart-session/utils/CosignerService.ts

```diff
@@ -38,26 +37,67 @@ export async function sendCoSignerRequest<
   transformRequest?: (data: TRequest) => unknown
 }): Promise<TResponse> {
   try {
+    // Transform request data if a transform function is provided
     const transformedData = transformRequest ? transformRequest(request) : request
-    const response = await axios.post<TResponse>(url, transformedData, {
-      params: queryParams,
-      headers
-    })
 
-    return response.data
+    // Construct query string
+    const queryString = new URLSearchParams({
+      ...queryParams
+    }).toString()
+
+    const fullUrl = queryString ? `${url}?${queryString}` : url
+
+    // Prepare fetch options
+    const fetchOptions: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        ...headers
+      },
+      body: JSON.stringify(transformedData)
+    }
+
+    // Perform the fetch request
+    const response = await fetch(fullUrl, fetchOptions)
+
+    // Check if the response is ok (status in the range 200-299)
+    if (!response.ok) {
+      // Try to parse error response
+      const errorData = await response.text()
+      throw new CoSignerApiError(response.status, errorData)
+    }
+
+    // Handle void responses explicitly
+    const contentLength = response.headers.get('content-length')
+    const contentType = response.headers.get('content-type')
+
+    // If response is empty or has no content
+    if (
+      response.status === 204 ||
+      contentLength === '0' ||
+      !contentType ||
+      !contentType.includes('application/json')
+    ) {
+      return undefined as TResponse
+    }
```

Comment:
If we are going to use this logics in different requests, can we move it to a named function to encapsulate the logic for readability and reusability? 

For example:
```js
function isResponseEmptyOrInvalid(response: Response, contentType: string | null, contentLength: string | null): boolean {
  return (
    response.status === 204 || // No content
    contentLength === '0' || // Explicit zero-length
    !contentType || // No content type
    !contentType.includes('application/json') // Non-JSON response
  );
}
...
if (isResponseEmptyOrInvalid(response, contentType, contentLength)) {
  return undefined as TResponse;
}
```